chia/data_layer/data_store.py

```diff
@@ -491,40 +596,72 @@ async def get_node_type(self, node_hash: bytes32, *, lock: bool = True) -> NodeT
 
         return NodeType(raw_node_type["node_type"])
 
+    async def get_terminal_node_for_seed(
+        self, tree_id: bytes32, seed: bytes32, *, lock: bool = True
+    ) -> Optional[bytes32]:
+        path = int(seed.hex(), 16)
+        async with self.db_wrapper.locked_transaction(lock=lock):
+            root = await self.get_tree_root(tree_id, lock=False)
+            if root is None or root.node_hash is None:
+                return None
+            node_hash = root.node_hash
+            while True:
+                node = await self.get_node(node_hash, lock=False)
+                assert node is not None
+                if isinstance(node, TerminalNode):
+                    break
+                if path % 2 == 0:
+                    node_hash = node.left_hash
+                else:
+                    node_hash = node.right_hash
+                path = path // 2
```

Comment:
this violated the `assert len(ancestors) <= 65` restriction for some reason. it seems not uniform for some reason, tho I'm not sure why. note that seed is passed as a `std_hash` of key and value, so it should be completely random where they end up.

Reply:
say bytes is .....001001. then we go right, left, left, right, left, left and so on. only one change in a bit will produce a different path.

Reply:
Yeah, I misreasoned about the present behavior.  Thanks for correcting me.  I guess if you end up with a high hash on a node at the top it will shove everything left.  @emlowe, this was your idea.  :point_left: Maybe I'm not thinking about it right.  We would want something where for any given hash, half of all hashes would end up left of it.

Either way, I think that the same logic should be used here as in the other place I commented on the decision making process.

Reply:
well if we want to include both `node_hash` and `seed` into the equation and we're at i-th height we can compare i-th bit of `node_hash` with i-th bit of `seed`. if they are equal we go left, otherwise we go right.

but is there any reason to include `node_hash` into the path decision? I think `seed` alone gives enough randomness, thanks to `std_hash` from `Program.to(`